src/cloudant/result.py

```diff
@@ -347,21 +348,47 @@ def __iter__(self):
         except ValueError:
             raise ResultException(104, self._page_size)
 
-        skip = 0
+        initial_options = {option: value
```

Comment:
Not sure about this. Bit tricky to read. How about...
```py
        _call = partial(self._ref, limit=self._page_size + 1, **self.options)

        init_opts = {
            'skip': self.options.pop('skip', None),
            'startkey': self.options.pop('startkey', None)
        }

        response = _call(**{k: v for k, v in init_opts.items() if v is not None})
```

Reply:
ok, it's more concise and definitely easier to read!